desc/input_reader.py

```diff
@@ -558,20 +563,20 @@ def parse_inputs(self, fname=None):  # noqa: C901 - FIXME: simplify this
         if curr_flag and iota_flag:
             raise OSError(colored("Cannot specify both iota and current.", "red"))
 
-        # remove unused profile
-        if iota_flag:
-            if inputs["objective"] != "vacuum":
-                del inputs["current"]
-            else:  # if vacuum objective from input file, use zero current
-                del inputs["iota"]
-        else:
-            del inputs["iota"]
-
-        if inputs["objective"] == "vacuum" and (pres_flag or iota_flag or curr_flag):
+        if vac_flag and (pres_flag or iota_flag or curr_flag):
             warnings.warn(
                 "Vacuum objective does not use any profiles, "
```

Comment:
Is it better to say that the vacuum objective flag assumes p=0 and I=0 as the profiles? Since we have changed it to now just be force balance with 0 p and 0 I?